scripts/generate-cmake-files.js

```diff
@@ -18,8 +22,67 @@ function generateFileList(name, regex, patterns) {
         groups[group].push(file);
     }
 
-    const fileListCmake = ejs.compile(fs.readFileSync('cmake/files.txt.ejs', 'utf8'), {strict: true});
-    writeIfModified(`cmake/${name}-files.txt`, fileListCmake({ groups: groups }));
+    writeIfModified(`cmake/${name}-files.txt`, fileListTxt({ groups: groups }));
+}
+
+function generateXcodeSourceList(project, target, name) {
+    const projectPath = path.dirname(project);
+    let objects = xcode.project(path.join(project, 'project.pbxproj')).parseSync().hash.project.objects;
+
+    // Build reverse lookup table for file => group
+    let objToGroup = {};
+    Object.keys(objects.PBXGroup)
+        .filter(name => !/_comment$/.test(name))
+        .forEach(name => objects.PBXGroup[name].children
+            .forEach(child => objToGroup[child.value] = name));
+
+    // Retrieve the target
+    const targetObj = Object.keys(objects.PBXNativeTarget)
+        .filter(name => !/_comment$/.test(name))
+        .map(name => objects.PBXNativeTarget[name])
+        .filter(project => project.name == target)[0];
+
+    // Retrieve the sources associated with the target.
+    const sourcesObj = targetObj.buildPhases
+        .map(phase => objects.PBXSourcesBuildPhase[phase.value])
+        .filter(phase => phase)[0];
```

Comment:
Does the list need to also include headers and resources? If it includes resources, there might need to be logic around file variants (for localized versions of files).

Reply:
Thanks, I added the headers as well.